torch/csrc/distributed/c10d/init.cpp

```diff
@@ -1529,6 +1543,22 @@ static const auto ProcessGroupNCCLOptionsTorchBind =
 
 static const auto ProcessGroupNCCLTorchBind =
     torch::class_<::c10d::ProcessGroupNCCL>("dist_c10d", "ProcessGroupNCCL")
+        .def_pickle(
+            [](const c10::intrusive_ptr<::c10d::ProcessGroupNCCL>& self) {
+              auto base_process_group =
+                  static_cast<c10::intrusive_ptr<::c10d::ProcessGroup>>(self);
+              auto name =
+                  ::c10d::DistributedC10d::get()->getNameOfProcessGroup(self);
+              return std::vector<std::string>{name};
+            },
+            [](std::vector<std::string> state) {
+              TORCH_CHECK(
+                  state.size() == 1,
+                  "Expecting exactly 1 state when restoring ProcessGroupNCCL, got: ",
+                  state.size());
+              auto base_process_group = ::c10d::DistributedC10d::get()->getProcessGroupByName(state.front());
```

Comment:
what if we only get a NullProcessGroup here? shouldn't this consider as not successfully reloaded as the previous ProcessGroup already "disappear"?